runtime_services/src/runtime_services.rs

```diff
@@ -0,0 +1,864 @@
+#![cfg_attr(all(not(test), not(feature = "mockall")), no_std)]
+
+extern crate alloc;
+
+pub mod variable_services;
+
+#[cfg(any(test, feature = "mockall"))]
+use mockall::automock;
+
+use alloc::vec::Vec;
+use core::{
+    ffi::c_void,
+    marker::PhantomData,
+    ptr,
+    sync::atomic::{AtomicPtr, Ordering},
+};
+
+use r_efi::efi;
+pub use variable_services::{GetVariableStatus, VariableIdentifier, VariableInfo, VariableNameIterator};
+
+/// This is the runtime services used in the UEFI.
+/// it wraps an atomic ptr to [`efi::RuntimeServices`]
+#[derive(Debug)]
+pub struct StandardRuntimeServices<'a> {
+    efi_runtime_services: AtomicPtr<efi::RuntimeServices>,
+    _lifetime_marker: PhantomData<&'a efi::RuntimeServices>,
+}
+
+impl<'a> StandardRuntimeServices<'a> {
+    /// Create a new StandardRuntimeServices with the provided [efi::RuntimeServices].
+    pub const fn new(efi_runtime_services: &'a efi::RuntimeServices) -> Self {
+        // The efi::RuntimeServices is only read, that is why we use a non mutable reference.
+        Self {
+            efi_runtime_services: AtomicPtr::new(efi_runtime_services as *const _ as *mut _),

```

Comment:
I think the preference is to use `addr_of_mut!` for pointer work. Can you see if that works here?

https://doc.rust-lang.org/stable/core/ptr/macro.addr_of_mut.html#

Reply:
I don't believe I would be able to use that here since that would produce a `*mut &'a r_efi::system::RuntimeServices`, instead of a `*mut r_efi::system::RuntimeServices`, unless I'm mistaken.

Reply:
I was not positive. If that is what you saw when you attempted the change, then I believe you. Thanks for checking :)